File: src/api/Dtos/OrderDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Example.Api.Dtos;

/// <summary>
/// The data transfer object for an patient order.
/// </summary>
/// <value></value>
public record OrderDto
{
    /// <summary>
    /// Gets the unique identifier for the order.
    /// </summary>
    public long Id { get; init; }

    /// <summary>
    /// Gets the message associated with the order.
    /// </summary>
    public string Message { get; init; } = string.Empty;

    /// <summary>
    /// The patient id related to the order.
    /// </summary>
    /// <value></value>
    public long PatientId { get; init; }

    /// <summary>
    /// Gets the date and time when the order was created.
    /// </summary>
    public DateTimeOffset CreatedAt { get; init; }
}

File: src/nodejs/src/dtos/OrderDto.ts
/**
 * Data Transfer Object for order information.
 */
export class OrderDto {
  /** The unique identifier of the order. */
  id: number;
  /** The message associated with the order. */
  message: string;
  /** The ID of the patient who owns the order. */
  patientId: number;
  /** The timestamp when the order was created. */
  createdAt: Date;
  /** The timestamp when the order was last updated. */
  updatedAt?: Date;

  /**
   * Creates a new order DTO.
   * @param id - The unique identifier of the order.
   * @param message - The message associated with the order.
   * @param patientId - The ID of the patient who owns the order.
   * @param createdAt - The timestamp when the order was created.
   * @param updatedAt - The timestamp when the order was last updated.
   */
  constructor(id: number, message: string, patientId: number, createdAt: Date, updatedAt?: Date) {
    this.id = id;
    this.message = message;
    this.patientId = patientId;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
  }
}
